connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```
@@ -1803,12 +1953,52 @@ ble_error_t Gap::setExtendedAdvertisingParameters(
 )
 {
+    tr_info("Set extended advertising parameters for set %d: "
+            "_advType=%s "
+            "_minInterval=%lums "
+            "_maxInterval=%lums "
+            "_peerAddressType=%s "
+            "_ownAddressType=%s "
+            "_policy=%s "
+            "_primaryPhy=%s "
+            "_secondaryPhy=%s "
+            "_peerAddress=%s "
+            "_txPower=%d "
+            "_maxSkip=%u "
+            "_channel37=%s "
+            "_channel38=%s "
+            "_channel39=%s "
+            "_anonymous=%s "
+            "_notifyOnScan=%s"
+            "_legacyPDU=%s "
+            "_includeHeaderTxPower=%s",
+            handle,
+            to_string(params.getType()),
+            params.getMinPrimaryInterval().valueInMs(),
+            params.getMaxPrimaryInterval().valueInMs(),
+            to_string(params.getPeerAddressType()),
+            to_string(params.getOwnAddressType()),
+            to_string(params.getFilter()),
+            to_string(params.getPrimaryPhy()),
+            to_string(params.getSecondaryPhy()),
+            to_string(params.getPeerAddress()),
+            params.getTxPower(),
+            params.getSecondaryMaxSkip(),
+            to_string(params.getChannel37()),
+            to_string(params.getChannel38()),
+            to_string(params.getChannel39()),
+            to_string(params.getAnonymousAdvertising()),
+            to_string(params.getScanRequestNotification()),
+            to_string(params.getUseLegacyPDU()),
+            to_string(params.getTxPowerInHeader()));
+
     /* check for illegal parameter combination */
     if ((params.getType() == advertising_type_t::CONNECTABLE_UNDIRECTED ||
          params.getType() == advertising_type_t::CONNECTABLE_DIRECTED) &&
         params.getUseLegacyPDU() == false
     ) {
         /* these types can only be used with legacy PDUs */
+        tr_error("Failed: directed and undirected connections can only be used with legacy PDUs");
```

Bellow an error is not handled, you should add an else after the `if(!err)`